File: sort/insertion_sort/go/insertion_sort.go
package main

import "fmt"

func insertionSort(arr []float64) []float64 {
	var key float64
	var i int

	for j := 0; j < len(arr); j++ {
		key = arr[j]

		i = j - 1

		for i > 0 && arr[i] > key {
			arr[i+1] = arr[i]
			i--
		}
		arr[i+1] = key
	}

	return arr
}

func main() {
	var n int

	fmt.Printf("Enter number of integers : ")
	fmt.Scan(&n)

	arr := make([]float64, n)

	fmt.Printf("\nEnter array elements : ")

	for i := 0; i < n; i++ {
		fmt.Scan(&arr[i])
	}

	arr = insertionSort(arr)
	fmt.Printf("Sorted array is ")

	for i := 0; i < n; i++ {
		fmt.Printf("%v ", arr[i])
	}
}

File: sort/insertion_sort/cpp/insertionSort.cpp
#include <iostream>
using namespace std;
int *insertionSort(int unsortedArray[], int length)
{
    for (int i = 1; i < length; i++)
    {
        int j = i;
        int value = unsortedArray[i];
        while (j >= 0 && (value < unsortedArray[j - 1]))
        {
            unsortedArray[j] = unsortedArray[j - 1];
            j--;
        }
        unsortedArray[j] = value;
    }
    return unsortedArray;
}
int main()
{
    int length;
    cout << "enter the number of elements want to sort" << endl;
    cin >> length;
    int unsortedArray[length];
    for (int i = 0; i < length; i++)
    {
        cin >> unsortedArray[i];
    }
    int *sortedArray = insertionSort(unsortedArray, length);
    for (int i = 0; i < length; i++)
    {
        cout << sortedArray[i] << endl;
    }
}